src/lua/msgpack.c

```diff
@@ -646,6 +646,15 @@ luamp_new_object(struct lua_State *L, size_t data_len)
 	return obj;
 }
 
+void
+luamp_new_object(struct lua_State *L, const char *data, const char *data_end)
```

Comment:
I don't want to expose `struct luamp_object` outside `msgpack.c`, because users need only a pointer to msgpack data. I think the `luamp_` prefix is better than `luaT_` in out case, because it's consistent with other functions from `msgpack.h` (`luamp_decode`, `luamp_encode`). Also, I think that the `_object` suffix is redundant in the API.

So I renamed `luamp_new_object` to `luamp_push` and `luamp_is_object` to `luamp_get`, and dropped the `_internal` suffix from `luamp_new_object`.